edge-modules/DirectMethodSender/Program.cs

```diff
@@ -21,38 +21,55 @@ public static async Task<int> MainAsync()
         {
             Logger.LogInformation($"Starting DirectMethodSender with the following settings:\r\n{Settings.Current}");
 
+            ModuleClient moduleClient = null;
             try
             {
-                ModuleClient moduleClient = await ModuleUtil.CreateModuleClientAsync(
+                (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
+
+                moduleClient = await ModuleUtil.CreateModuleClientAsync(
                     Settings.Current.TransportType,
                     ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
                     ModuleUtil.DefaultTransientRetryStrategy,
                     Logger);
 
-                Uri analyzerUrl = Settings.Current.AnalyzerUrl;
-                AnalyzerClient analyzerClient = new AnalyzerClient { BaseUrl = analyzerUrl.AbsoluteUri };
-
-                (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
+                Option<Uri> analyzerUrl = Settings.Current.AnalyzerUrl;
+                if (analyzerUrl.HasValue)
+                {
+                    analyzerUrl.ForEach(async (uri) =>
+                    {
+                        AnalyzerClient analyzerClient = new AnalyzerClient { BaseUrl = uri.AbsoluteUri };
+                        Action<MethodResponse> reportResult = async (response) => await ReportStatus(Settings.Current.TargetModuleId, response, analyzerClient);
+                        await StartDirectMethdTests(moduleClient, reportResult, Settings.Current.DirectMethodDelay, cts);
+                    });
+                }
+                else
+                {
+                    Action<MethodResponse> reportResult = async (response) => await moduleClient.SendEventAsync("AnyOutput", new Message(Encoding.UTF8.GetBytes("Direct Method call succeeded.")));
```

Comment:
How will this message get processed? Is this existing logic?

Reply:
Yes, existing logic in e2e